src/persistence/settingsserializer.cpp

```diff
@@ -180,9 +188,12 @@ const SettingsSerializer::Value* SettingsSerializer::findValue(const QString& ke
         {
             if (a.group != group)
                 continue;
-            for (int vi : a.values)
-                if (values[vi].arrayIndex == arrayIndex && values[vi].key == key)
-                    return &values[vi];
+            for (int vi = 0; vi < a.values.size(); vi++)
+            {
+                const Value& v = values[vi];
+                if (v.arrayIndex == arrayIndex && v.key == key)
+                    return &v;
+            }
```

Comment:
This completely changes the meaning of the loop! You're now iterating over all values, instead of the values of the array.
What was wrong with the simple range-for loop?